### PR TITLE
Dependencies updated

### LINKED ISSUE #128: Output Window gets opened when I type
Output window gets opened any time I type some code and shows language service output.

When I close it doesn't have permanent effect, it gets opened again when I type.

I can't see any settings in the plugin. What shall I do? I don't want Output window open.

### PR DESCRIPTION
**Pull request by DBosley**
All major dependencies have been updated to latest except for deps related to this repo: https://github.com/Microsoft/vscode-languageserver-node

There seems to be a bug with piping data via stdio when using version 3.3.0+ of the above package and I've submitted an issue with it:
https://github.com/Microsoft/vscode-languageserver-node/issues/229

All other deps have been updated and braking changes accounted for.
The reason for the large number of changes was because I was trying to determine the cause of the failed TravisCI builds. They were all related to the 3.3.0+ versions of vscode-laungageserver-node dependencies. After forcing them all to be "~3.2.0" the integration tests passed after updating the golden outputs.

Local language-server debugging for this project has been fixed. It now more closely matches the example project from Microsoft.

This PR resolves #128 and possibly others.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -18,24 +18,25 @@
   ],
   "main": "./out/extension",
   "contributes": {
-    "languages": [
```
Comment (chuckjaz) [inline client/package.json]: This formatting changes appears unnecessary.

File: client/package.json [hunk 2]
```diff
@@ -50,4 +51,4 @@
     "type": "git",
     "url": "https://github.com/angular/vscode-ng-template-service"
   }
-}
```
Comment (chuckjaz) [inline client/package.json]: Please ensure the file ends in a blank line.

File: tests/assets/startup-golden.json [hunk 3]
```diff
@@ -1 +1 @@
-[{"jsonrpc":"2.0","id":2,"result":null}]
+[{"jsonrpc":"2.0","id":2,"result":null}]
```
Comment (chuckjaz) [inline tests/assets/startup-golden.json]: Revert this change.

File: tests/assets/completion-golden.json [hunk 4]
```diff
@@ -1 +1 @@
...ext":"startsWith","insertText":"startsWith"},{"label":"anchor","kind":2,"detail":"method","sortText":"anchor","insertText":"anchor"},{"label":"big","kind":2,"detail":"method","sortText":"big","insertText":"big"},{"label":"blink","kind":2,"detail":"method","sortText":"blink","insertText":"blink"},{"label":"bold","kind":2,"detail":"method","sortText":"bold","insertText":"bold"},{"label":"fixed","kind":2,"detail":"method","sortText":"fixed","insertText":"fixed"},{"label":"fontcolor","kind":2,"detail":"method","sortText":"fontcolor","insertText":"fontcolor"},{"label":"fontsize","kind":2,"detail":"method","sortText":"fontsize","insertText":"fontsize"},{"label":"italics","kind":2,"detail":"method","sortText":"italics","insertText":"italics"},{"label":"link","kind":2,"detail":"method","sortText":"link","insertText":"link"},{"label":"small","kind":2,"detail":"method","sortText":"small","insertText":"small"},{"label":"strike","kind":2,"detail":"method","sortText":"strike","insertText":"strike"},{"label":"sub","kind":2,"detail":"method","sortText":"sub","insertText":"sub"},{"label":"sup","kind":2,"detail":"method","sortText":"sup","insertText":"sup"}]},{"jsonrpc":"2.0","id":2,"result":null}]
```
Comment (chuckjaz) [inline tests/assets/completion-golden.json]: Revert this change.

**Comment by chuckjaz** [comment]
Can you also squash the commits and include a `Fixes: #xxxxx` for any issues that are resolved by the commit.

**Comment by DBosley** [comment]
@chuckjaz I've made the requested changes. This PR only fixes #128 for error cases already fixed in the @angular/language-service project. I'm looking into ways to fix this for any and all errors next

File: client/src/extension.ts [hunk 5]
```diff
@@ -7,13 +7,13 @@ export function activate(context: ExtensionContext) {
...
   let serverModule = context.asAbsolutePath(path.join('server', 'server.js'));
   // The debug options for the server
-  let debugOptions = { execArgv: ["--nolazy", "--debug=6004"] };
+  let debugOptions = { execArgv: ["--nolazy", "--debug=6009"] };
 
   // If the extension is launched in debug mode then the debug server options are used
   // Otherwise the run options are used
   let serverOptions: ServerOptions = {
     run : { module: serverModule, transport: TransportKind.ipc },
-    debug: { module: serverModule, transport: TransportKind.ipc /* *, options: debugOptions /* */ }
```
Comment (chuckjaz) [inline client/src/extension.ts]: Please add the comments back for commit to master. They should only be removed during testing. Also note that simply adding a single '/' after the second start has the same effect as deleting the comments. That explains the funny formatting of the comment (http://removingalldoubt.com/programming/2005/09/03/Comment-Toggle)

File: tests/tools/validate-expected/index.ts [hunk 6]
```diff
@@ -87,7 +87,6 @@ if (!isGolden && !expectedFile) {
 }
 
 const reader = new rpc.StreamMessageReader(process.stdin);
-
```
Comment (chuckjaz) [inline tests/tools/validate-expected/index.ts]: Please revert this change. I like my commits to have a single purpose. If you feel this formatting change is necessary please either submit a follow-up formatting PR or split the formatting changes into a separate commit in this PR.

File: client/.vscode/tasks.json [hunk 7]
```diff
@@ -0,0 +1,30 @@
+// Available variables which can be used inside of strings.
```
Comment (chuckjaz) [inline client/.vscode/tasks.json]: This file seems unnecessary to the fix. See my note below about single purpose commits.

**Comment by chuckjaz** [comment]
I believe this is very close. I will cut a new release after this lands.

File: client/src/extension.ts (hunk 5 above)
Comment (DBosley) [inline client/src/extension.ts]: @chuckjaz As I understood it, the debug options do not affect the extension when it's running without the debug flag. I'm happy to add the comment-toggle back in, but I was curious to the reasoning behind this decision.

File: tests/tools/validate-expected/index.ts (hunk 6 above)
Comment (DBosley) [inline tests/tools/validate-expected/index.ts]: Made this change purely by habit. Will revert.

File: client/.vscode/tasks.json (hunk 7 above)
Comment (DBosley) [inline client/.vscode/tasks.json]: This was added to more closely follow the Microsoft example app. They use a preLaunchTask run the compile-watch command before the debugger begins.
I had the changes that lined the extension up more closely to their example in one commit originally, but they were all squashed with your request. Do you want me to submit the changes that line up the app to the example as a separate PR?

**Comment by DBosley** [comment]
@chuckjaz I've reverted all the non-dependency related changes that fixed local debugging and submitted them as a separate PR #139.
